### PR TITLE
Updating docs

### PR DESCRIPTION
**Pull request by eliasdorneles**
Hey, folks!

Here is an update proposal for the docs, please let me know what you think.

Summary of changes:

* introduces `dateparser.parse()` as main entry point (friendlier to new users)
* added more details in DateDataParser description
* updated installation section and Scrapy Cloud deploy to use `shub deploy-egg`
* updated autogenerated docs

Does this look good?

### CODE REVIEW

**Comment by Allactaga** [comment]
Hey Elias, thanks for covering the docs. We are at the end of deadline ATM, but we'll look at this soon as we can (most likely Waqas).

I had a discussion one day with one of developers outside of our team, and he told me that following our docs is not enough to deploy dateparser into a Scrapy Cloud project. You'd also need to deploy `dateutil` and `PyYAML` eggs. Also I have this command in my history which I think is related to `PyYAML` egg, but I don't remember details: `python setup.py --without-libyaml bdist_egg`. Could you please follow up on this issue?

**Comment by eliasdorneles** [comment]
Right, I'll look into that.

File: docs/usage.rst [hunk 1]
```diff
@@ -20,16 +69,12 @@ Here is a quick example of usage::
     >>> ddp.get_date_data('13 August, 2014')
     {'date_obj': datetime.datetime(2014, 8, 13, 0, 0), 'period': 'day'}
 
+Note: the possible values for `'period'` are currently: `'years'`, `weeks` or `day`.
```
Comment (waqasshabbir) [inline docs/usage.rst]: It's `year` and `month` without `s`.

Comment (eliasdorneles) [inline docs/usage.rst]: oops, good catch, @waqasshabbir -- thank you! :)

**Comment by eliasdorneles** [comment]
Hey fellows, I've just updated this with instructions to deploy the dependencies and fixing the typos pointed by Waqas.

Only one thing: this is pointing to the latest version of shub which isn't yet released on PyPI.
I will release it later today (have to first fix a bug I found while testing the instructions here), I can leave a note here after.

**Comment by eliasdorneles** [comment]
Hey, fellows -- new shub is released, grab it while it's hot! :)

Sorry for the delay, it took a bit longer than I thought to sort out the issues, but now we can merge this PR if it looks good.

**Comment by waqasshabbir** [comment]
Hi @eliasdorneles I've borrowed your changes from this PR to issue PR #89. Let's review it together.

**Pull request merged**